### PR TITLE
Feature: Add config setting to disable gzip compression #1627

### PR DESCRIPTION
**Pull request by markdingram**
- rounds out all of the https://kubernetes.io/docs/reference/config-api/kubeconfig.v1/#Cluster options

### CODE REVIEW

File: kube-client/src/client/builder.rs [hunk 1]
```diff
@@ -158,7 +158,13 @@ where
     #[cfg(feature = "gzip")]
     let stack = ServiceBuilder::new()
         .layer(stack)
-        .layer(tower_http::decompression::DecompressionLayer::new())
+        .layer(
+            tower_http::decompression::DecompressionLayer::new()
+                .no_br()
```
Comment (markdingram) [inline kube-client/src/client/builder.rs:163]: intentionally disable anything but GZIP compression - 

reading https://github.com/kubernetes/kubernetes/issues/112296 show there isn't like to be any other option in Kube for a while, but these flags could easily be refitted when/if Kube gains alternatives.

File: kube-client/src/client/builder.rs [hunk 2]
```diff
@@ -158,7 +158,13 @@ where
     #[cfg(feature = "gzip")]
     let stack = ServiceBuilder::new()
         .layer(stack)
-        .layer(tower_http::decompression::DecompressionLayer::new())
+        .layer(
```
Comment (markdingram) [inline kube-client/src/client/builder.rs:161]: trying to wrap the whole layer in an "option_layer" block ran head first into a load of compile issues - fortunately the Compression Layer supports enabling/disabling through configuration hence that path was chosen

**Comment by markdingram** [comment]
Remaining work on this PR is to add a test to check the Accept-Encoding header when the disabled flag is set

File: kube-client/src/client/builder.rs (hunk 1 above)
Comment (clux) [reply kube-client/src/client/builder.rs:163]: agreed. think this is sensible also.

File: kube-client/src/client/builder.rs [hunk 3]
```diff
@@ -226,3 +232,70 @@ where
...
+        assert_eq!(&response, "gzip");
+
+        // now disable and check empty string echoed back
+        let config = Config {
+            disable_compression: true,
+            ..config
+        };
+        let client = make_generic_builder(HttpConnector::new(), config)?.build();
+        let response = client.request_text(http::Request::default()).await?;
+        assert_eq!(&response, "");
```
Comment (clux) [inline kube-client/src/client/builder.rs:297]: wow great test. 👍

**Review by clux — APPROVED** [review]
Looks great! Thank you very much. Sensible porting of upstream behavior and sensible defaulting.

we've not historically been great at testing this area of the code base and this is a very nice way of doing it

**Comment by clux** [comment]
Sorry for the late merge. Will get this released soon!

**Pull request merged**